src/i2c.rs

```diff
@@ -0,0 +1,217 @@
+use std::marker::PhantomData;
+
+use crate::bindings::{
+    CgosI2CCount, CgosI2CGetFrequency, CgosI2CGetMaxFrequency, CgosI2CIsAvailable, CgosI2CRead,
+    CgosI2CReadRegister, CgosI2CSetFrequency, CgosI2CType, CgosI2CWrite, CgosI2CWriteReadCombined,
+    CgosI2CWriteRegister, CGOS_I2C_TYPE_DDC, CGOS_I2C_TYPE_PRIMARY, CGOS_I2C_TYPE_SMB,
+    CGOS_I2C_TYPE_UNKNOWN,
+};
+
+/// Error type for I2c operations
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub enum I2cErr {
+    /// User-supplied index is out of range
+    IdxOutOfRange,
+
+    /// I2c bus transaction failed
+    BusErr,
+}
+
+pub type I2cResult<T> = Result<T, I2cErr>;
```

Comment:
```suggestion
pub type Result<T> = Result<T, I2cErr>;
```

this type is already inside the `i2c` module namespace, do not repeat yourself in the naming